report/report.go

```diff
@@ -409,6 +426,36 @@ func (r Report) upgradePodNodes() Report {
 	return r
 }
 
+func (r Report) upgradeNamespaces() Report {
+	if len(r.Namespace.Nodes) > 0 {
+		return r
+	}
+
+	namespaces := map[string]struct{}{}
+	for _, t := range []Topology{r.Pod, r.Service, r.Deployment, r.DaemonSet, r.StatefulSet, r.CronJob} {
+		for _, n := range t.Nodes {
+			if state, ok := n.Latest.Lookup(KubernetesState); ok && state == KubernetesStateDeleted {
+				continue
+			}
+			if namespace, ok := n.Latest.Lookup(KubernetesNamespace); ok {
+				namespaces[namespace] = struct{}{}
+			}
+		}
+	}
+
+	nodes := Nodes{}
+	for ns := range namespaces {
+		// Namespace ID
+		// Probes did not use to report namespace ids, but since creating a report node requires an id,
+		// the namespace name, which is unique, is passed to `MakeNamespaceNodeID
+		namespaceID := MakeNamespaceNodeID(ns)
+		nodes[namespaceID] = MakeNodeWith(namespaceID, map[string]string{KubernetesName: ns})
```

Comment:
Oh, is that because "proper" namespace nodes do get field courtesy of `meta.MetaNode()`? If so, should we be adding a `KubernetesNamespace` key too, since `meta.MetaNode()` adds that too?

It's a bit silly to have the name in three places, but probably not worth breaking the pattern for that.